README.md

```diff
@@ -187,46 +189,63 @@ Please examine the XML docs for full details. However, in simplified form, we ha
 
     public class Egg
     {
+        public Egg()
+        {
+            Id = 3;
+            Name = "Third";
+        }
         public int Id { get; set; }
         public string Name { get; set; }
     }
 
-    [BuilderFactory]
     public static class MonsterBuilder
     {
-        public static Builder<Monster> New
+        public static Builder<Monster> Default
         {
             get
             {
                 var rand = new Random();
+                var theBiggestLake = new Lake();
+
                 return Builder<Monster>.New
-                    .With(m => m.Nationality, "Scottish")                                   // All monsters will be 'Scottish'
+                    .With(m => m.Nationality, "Scottish")                                               // All monsters will be Scottish.
+
+                    .WithSequentialIds(m => m.Id)                                                       // Ids will be 1, 2, 3, 4, 5....
+                                                                                                        // Above is identical to ".WithSequentialFrom(t => t.Id, Enumerable.Range(1, int.MaxValue))"
 
-                    .WithSequentialIds(m => m.Id)                                           // Ids will be 1, 2, 3, 4, 5....
-                    // Above is identical to ".With(t => t.Id, Enumerable.Range(1, int.MaxValue))"
+                    .WithSharedRef(m => m.CommunityLake, theBiggestLake)                                // All monsters will have a reference to the same Lake in this.CommunityLake.
 
-                    .WithSequentialFrom(m => m.Colour, "Green", "Red", "Blue")              // Monster Colors will be Green, Red, Blue, Green, Red, ...
+                    .WithSequentialFrom(m => m.Colour, "Green", "Red", "Blue")                          // Monster Colors will be Green, Red, Blue, Green, Red, ...
 
-                    .WithCreateEnumerableFrom(m => m.Sounds, "Rarrrgggh!", "Screech!", "Wooooosh!") // All monsters will produce ALL three of these sounds.
-                    // Above is identical to ".WithCreateEnumerableFrom(m => m.Sounds, new List<string>{"Rarrrgggh!", "Screech!", "Woooooh!"})"
-                    // Above is *almost* identical to ".With(m => m.Sounds, new []{"Rarrrgggh!", "Screech!", "Woooooh!"})". The only difference is that the containing array is not shared.
+                    .WithCreateEnumerableFrom(m => m.Sounds, "Rarrrgggh!", "Screech!", "Wooooosh!")     // All monsters will produce all three of these sounds.
+                                                                                                        // Above is identical to ".WithCreateEnumerableFrom(m => m.Sounds, new List<string>{"Rarrrgggh!", "Screech!", "Woooooh!"})"
```

Comment:
This comment is self-referential, I'd probably prefer:
```
// This is equivalent to ".WithCreateEnumerableFrom(m => m.Sounds, new List<string>{"Rarrrgggh!", "Screech!", "Woooooh!"})"
// And to .WithFactory(m => m.Sounds, () => new List<string>{"Rarrrgggh!", "Screech!", "Wooooosh!"})
```

To be honest, I'm not sure I like the `WithCreateEnumerableFrom` method at all - I'd much rather have someone use `WithFactory` which is more obviously clear in what it does. I guess `WithCreateEnumerableFrom` can also be used to make shallow clones of the provided enumerable, so they don't share the same reference, but this feels a little unexpected - instead, they could simply use `.ToList()` to make the shallow clone (?)

Reply:
Hmm ... one of the main things it's doing is getting rid of irrelevant boilerplate of the details of the IEnumerable.
When you're setting up the test object you don't need to care about whether it's a List, Array or a ReadOnlyCollection - the library handles all that for you. But the docs here don't do a good job of making that clear.

I'll expand the comments to highlight that.

And if the user thinks it's not clear enough, they can just use the core method :)

Reply:
Done.